src/runtime_src/core/edge/sw_em/generic_pcie_hal2/halapi.cxx

```diff
@@ -107,7 +116,14 @@ int xclLoadXclBin(xclDeviceHandle handle, const xclBin *buffer)
   xclcpuemhal2::CpuemShim *drv = xclcpuemhal2::CpuemShim::handleCheck(handle);
   if (!drv)
     return -1;
-  return drv->xclLoadXclBin(buffer);
+  auto ret = drv->xclLoadXclBin(buffer);
+  if (!ret) {
+    //auto device = xrt_core::get_userpf_device(drv);
+    //device->register_axlf(buffer);
```

Comment:
This has to be enabled, otherwise XRT kernel APIs will not work and upstream code that relies on XML meta data from xclbin will not function correctly.

Reply:
@chvamshi-xilinx, can you provide your comment on this, is this required for sw emulation drivers?

Reply:
Hi @vboggara-xilinx , This change  is needed for XRT Kernel API's support. But there are other changes are also needs to be added for this. Please make these changes in your next PR.
@stsoe , This change is enabling KDS for sw_emu. @vboggara-xilinx  is going to send a new PR for XRT Kernel API's support

Reply:
@chvamshi-xilinx , noted it and will do necessary changes required for XRT kernel API's in my next publish.